webapp/src/ts/services/enketo.service.ts

```diff
@@ -473,12 +473,49 @@ export class EnketoService {
       .each((idx, element) => {
         const xpath = Xpath.getElementXPath(element);
         const $input: any = $('input[type=file][name="' + xpath + '"]');
-        const file = $input[0].files[0];
+        const inputElement = $input[0];
+        let file: File | undefined | null;
+
+        if (inputElement?.files?.length > 0 && inputElement.files[0]) {
+          file = inputElement.files[0];
+        } else {
+          file = convertDrawDataToFile(xpath);
```

Comment:
I haven't really looked into this at all but something reminded me of [a preexisting bug in this code](https://github.com/medic/cht-core/issues/8072). It might be worth looking into the `FileManager.getCurrentFiles();` function instead of using xpath.

Reply:
For future reference, after quite a bit of tinkering, the only way I found to leverage the functionality in FileManager while maintaining backwards compatibility with the way that we currently do attachment names is by making a small change to the FileManager code (this could be done via a patch or by coping+editing the FileManager code and supplying a custom `enketo/file-manager` value. Either way, we just need to expose the `name` value associated with each of the elements that has a file.  For example, we could set `file.xpath = this.name;` before pushing the file into the array in `getCurrentFiles`.  

Then, that would let us handle all file attachments here in the enketo service with code like this:

```ts
    const files = FileManager.getCurrentFiles();
    for (const file of files) {
      // replace instance root element node name with form internal ID
      const filename = 'user-file' +
        (file.xpath.startsWith('/' + doc.form) ? file.xpath : file.xpath.replace(/^\/[^/]+/, '/' + doc.form));
      this.attachmentService.add(doc, filename, file, file.type, !file.type);
    }
```

Reply:
I don't believe we need to change the file manager but we would need to handle old and new ways of naming attachments. This branch has my attempts from earlier in the year: https://github.com/medic/cht-core/compare/master...8072-repeat-field-inputs

It basically works but needs complete testing on device and unfortunately got deprioritised.

Reply:
> a change to the attachment name scheme is not a breaking change (as long as the app still functions as it did before)?

Yes I think so. When you say "the app still functions as it did before" we need to still render old reports correctly, so the rendering has to support both naming schemes indefinitely. I think that's ok as we work towards consistency.

I don't think it's a breaking change because I doubt anyone is relying on the file name, but I might be wrong. The problem is to support repeat files we _have_ to change the naming because otherwise they collide. One alternative is to only use the different naming scheme with repeats, but that's complicated and means we can't use enketo's filemanager.

On balance, I think we're ok to change the file name.

Reply:
Could/should we add a DB migration to go back and rename the existing attachments in the DB?  (Not sure exactly how we might balance the migration effort vs having to support both naming schemas indefinitely...)  It seems like it would be possible to write a migration, but it might trigger a lot of syncing....

Reply:
Yeah we basically don't do mass migrations any more. This is another example of something that could be hidden behind a webapp api so the Report type is standard, and old docs are converted on read so the complexity is buried deep down and not in the UI layer...